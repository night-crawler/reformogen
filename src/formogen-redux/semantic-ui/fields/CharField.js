import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { CharField as CharFieldComponent } from '~/formogen/semantic-ui/fields/CharField';

import { formId, fieldValue } from '~/formogen-redux/selectors';

import { storeFieldData } from '~/formogen-redux/actions';


function mapDispatchToProps(dispatch) {
  return { dispatch };
}

function mergeProps(stateProps, dispatchProps, ownProps) {
  const { dispatch } = dispatchProps;
  const { formId } = ownProps;

  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps,

    onChange: (e, { name, value }) => dispatch(storeFieldData(formId, name, value)),
    dispatch: undefined,
  };
}

export const CharField = connect(
  createStructuredSelector({
    formId,
    value: fieldValue
  }),
  mapDispatchToProps,
  mergeProps,
)(CharFieldComponent);
