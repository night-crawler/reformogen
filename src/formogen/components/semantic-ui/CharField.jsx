import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { Form } from 'semantic-ui-react';

import { errorsType, layoutOptsType } from '../fieldPropTypes';
import Label from './Label';

import { MessageList } from './MiscComponents';


CharField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.string,
    max_length: PropTypes.number,
    help_text: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    errors: errorsType,

    required: PropTypes.bool,
    editable: PropTypes.bool,

    helpTextOnHover: PropTypes.bool,
    layoutOpts: layoutOptsType,

    updateProps: PropTypes.func,
    onChange: PropTypes.func,
};
export default function CharField(props) {
    let _props = {
        name: props.name,
        value: props.value,
        placeholder: props.placeholder,
        onChange: props.onChange,
        type: 'text',
        maxLength: props.max_length,
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <Form.Input { ..._props } />
            { !props.helpTextOnHover ? <span className='help-text'>{ props.help_text }</span> : '' }
            <MessageList messages={ props.errors } />
        </Form.Field>
    );
}
