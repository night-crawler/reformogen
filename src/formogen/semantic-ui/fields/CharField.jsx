import React from 'react';
import PropTypes from 'prop-types';

import _ from 'lodash';

import { Form } from 'semantic-ui-react';

import { errorsType, layoutOptsType } from '../../fieldPropTypes';
import Label from '../common/Label';
import ErrorsList from '../common/ErrorsList';


CharField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.string,
    max_length: PropTypes.number,
    help_text: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    errors: errorsType,

    widget: PropTypes.string,
    password: PropTypes.bool,

    required: PropTypes.bool,
    editable: PropTypes.bool,

    helpTextOnHover: PropTypes.bool,
    layoutOpts: layoutOptsType,

    updateProps: PropTypes.func,
    onChange: PropTypes.func,
};
export default function CharField(props) {
    const isPassword = props.password || props.widget === 'password';

    let _props = {
        name: props.name,
        value: props.value,
        placeholder: props.placeholder,
        onChange: props.onChange,
        type: isPassword ? 'password' : 'text',
        maxLength: props.max_length,
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <Form.Input { ..._props } />
            {
                !props.helpTextOnHover
                    ? <span className='help-text'>{ props.help_text }</span>
                    : ''
            }
            <ErrorsList messages={ props.errors } />
        </Form.Field>
    );
}
