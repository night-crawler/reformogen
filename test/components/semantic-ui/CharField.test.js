import CharField from 'formogen/components/semantic-ui-fields/CharField';


it('should render CharField', () => {
    const wrapper = shallow(
        <CharField
            type='CharField'
            name='char_field_name'
            verbose_name="I've corkscrewed her twice for the last night!"
            help_text='Really?'
            layoutOpts={ { width: 1 } }
        />
    );
    expect(wrapper).toMatchSnapshot();
});
