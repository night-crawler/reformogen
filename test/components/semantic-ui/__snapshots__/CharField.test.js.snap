// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CharField 1`] = `
<FormField
  disabled={true}
  error={false}
  width={1}
>
  <SUILabel
    help_text="Really?"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="char_field_name"
    type="CharField"
    verbose_name="I've corkscrewed her twice for the last night!"
  />
  <FormInput
    as={[Function]}
    control={[Function]}
    name="char_field_name"
    type="text"
  />
  <span
    className="help-text"
  >
    Really?
  </span>
  <MessageList />
</FormField>
`;
